src/utils/useAsync.ts

```diff
@@ -0,0 +1,8 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import { useEffect } from "react";
+
+export const useAsync = (func: () => Promise<any>, inputs: any[]): void => {
```

Comment:
старайся обходится без `any`. тут можно типизировать так:
```suggestion
export const useAsync = (func: () => Promise<any>, inputs: React.DependencyList[]): void => {
```